tests/jax/components/building/adders_test.py

```diff
@@ -15,135 +15,211 @@
+    assert parallel_sequence_adder.name() == "executor_adder"
+
+
+def test_sequence_adders_signature(
+    mock_builder: Builder,
+    parallel_sequence_adder_signature: ParallelSequenceAdderSignature,
+) -> None:
+    """Test sequence adder signature callback.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder_signature: Fixture ParallelSequenceAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_sequence_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
+    )
+    assert mock_builder.store.adder_signature_fn
+    assert parallel_sequence_adder_signature.name() == "data_server_adder_signature"
+    assert issubclass(
+        parallel_sequence_adder_signature.config_class(),  # type: ignore
+        AdderSignatureConfig,
+    )
+
+
+def test_transition_adders(
+    mock_builder: Builder,
+    parallel_transition_adder: ParallelTransitionAdder,
+) -> None:
+    """Test transition adder callback
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_transition_adder: Fixture ParallelTransitionAdder.
+
+    Returns:
+        None
+    """
+    parallel_transition_adder.on_building_executor_adder(builder=mock_builder)
+    assert type(mock_builder.store.adder) == reverb_adders.ParallelNStepTransitionAdder
+    assert parallel_transition_adder.name() == "executor_adder"
+
+
+def test_transition_adders_signature(
+    mock_builder: Builder,
+    parallel_transition_adder_signature: ParallelTransitionAdderSignature,
+) -> None:
+    """Test transition adder signature callback
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_transition_adder_signature: Fixture ParallelTransitionAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_transition_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
     )
-    test_system_parallel_sequence_adder._builder.store.adder.add_first(env_restart)
-    test_system_parallel_sequence_adder._builder.store.server.stop()
+    assert mock_builder.store.adder_signature_fn
+    assert parallel_transition_adder_signature.name() == "data_server_adder_signature"
+    assert issubclass(
+        parallel_transition_adder_signature.config_class(),  # type: ignore
+        AdderSignatureConfig,
+    )
+
+
+def test_uniform_priority(
+    mock_builder: Builder,
+    uniform_priority: UniformAdderPriority,
+) -> None:
+    """Test uniform priority callback
 
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        uniform_priority: Fixture UniformAdderPriority.
 
-# TODO (Kale-ab): test adder behaviour in more detail
+    Returns:
+        None
+    """
+    uniform_priority.on_building_executor_adder_priority(builder=mock_builder)
+    assert mock_builder.store.priority_fns
+    uniform_priority.on_building_executor_adder_priority(builder=mock_builder)
```

Comment:
Why call it twice?

Reply:
not sure how the second one got it. Removed it

Reply:
It seems like the second call is still there at the moment.